vircadia/vircadia
# Material improvements

fixes the remaining issue with transparent shape entities mentioned here: https://github.com/vircadia/vircadia/issues/224#issuecomment-643830147
fixes https://github.com/vircadia/vircadia/issues/744
fixes https://github.com/vircadia/vircadia/issues/1085

support material entities on Gizmo entities, Image entities, and Text entities

caveats:
- Gizmos don't currently define UVs, which means any material textures or procedural shaders that use them won't work.  this is not hard to fix...but we have to define how we want those UVs to work.  do they act as a cutout, or do the UVs go along the 2D torus.  gizmos also don't currently respect material albedo/opacity, because they are controlled by multiple entity property parameters each
- Image entities still need a valid imageURL, even if you override their rendering completely via a material.  keepAspectRatio will still respect that original image
- Text entities only support materials on the background, not the text itself currently.  when this is later supported, it will be by targeting "mesh part" "1" instead of "0"

Test plan:
- follow the instruction in https://github.com/vircadia/vircadia/issues/224#issuecomment-643830147 to confirm that transparent shapes render correctly now
- run [this](https://gist.github.com/HifiExperiments/52a9ab66807756d3fbebf5820589c312/raw/5ef13e1110a3dbe82f71111eec87ef8f94b22637/VariousMaterialsAndEntitiesTest.js) script.  you will see 5 entities and 6 materials in front of you.
- press "u" to apply each material in turn to each entity.  they should behave as expected, notwithstanding the above caveats.  note that the 4th material will make the flat entities (text, image, gizmo) only render on one side because it defaults to a non-fallthrough cullFaceMode of CULL_BACK, so for example the gizmo will only be visible from underneath.
- verify that the create rotation circle things are now unlit (finally!)
